cmd/expand-pool.go

```diff
@@ -0,0 +1,561 @@
+		ticker := time.NewTicker(time.Second * 10)
+		for {
+			select {
+			case <-ticker.C:
+				lstat, err := os.Stat(configFile)
+				if err != nil {
+					continue
+				}
+				if lstat.ModTime().After(ltime) {
+					rd, err := Open(configFile)
+					if err != nil {
+						continue
+					}
+					defer rd.Close()
+					cf := &config.ServerConfig{}
+					dec := yaml.NewDecoder(rd)
+					dec.SetStrict(true)
+					if err = dec.Decode(cf); err != nil {
+						continue
+					}
+					globalServerCtxt.poolsRWMutex.Lock()
+					layout, err := buildDisksLayoutFromConfFile(true, cf.Pools, true)
+					if err != nil {
+						continue
+					}
+					ltime = lstat.ModTime()
+					if !reflect.DeepEqual(layout, ctxt.Layout) && len(layout.pools) != 0 {
+						fmt.Println("found diff here")
+						ticker.Reset(time.Hour)
+						poolEndpoints, _, err := CreatePoolEndpoints(globalMinioAddr, layout.pools...)
+						if err != nil {
+							continue
+						}
+						for i, endpointList := range poolEndpoints {
+							if err = globalEndpoints.Add(PoolEndpoints{
+								Legacy:       ctxt.Layout.legacy,
+								SetCount:     len(layout.pools[i].layout),
+								DrivesPerSet: len(layout.pools[i].layout[0]),
+								Endpoints:    endpointList,
+								Platform:     fmt.Sprintf("OS: %s | Arch: %s", runtime.GOOS, runtime.GOARCH),
+								CmdLine:      layout.pools[i].cmdline,
+							}); err != nil {
+								continue
+							}
+						}
+						zz, err := newErasureServerPools(GlobalContext, globalEndpoints)
+						if err != nil {
+							continue
+						}
+						z, ok := zz.(*erasureServerPools)
+						if ok {
+							err := z.Decommission(GlobalContext, 0)
+							if err != nil {
+								fmt.Println("Decommission", err)
+								continue
+							}
+							go func() {
+								for {
+									time.Sleep(time.Second)
+									status, err := z.Status(GlobalContext, 0)
+									if err != nil {
+										continue
+									}
+									if status.Decommission != nil && status.Decommission.Complete {
+										// when it's done
+										fmt.Println("Decommission complete")
+										// should rename
+										globalEndpoints.Remove(0)
+										newErasureServerPools(GlobalContext, globalEndpoints)
+										break
+									}
+								}
+							}()
+						}
+					}
+					globalServerCtxt.poolsRWMutex.Unlock()
+				}
+			case <-GlobalContext.Done():
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+const (
+	ExpandPoolsStatusApiPath = "/expand-pool/status"
+)
+
+type selfPoolExpandResponse struct {
+	selfPoolExpand
+	Success bool `json:"success"`
+}
+
+func (a adminAPIHandlers) ExpandPoolsStatus(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	objectAPI, _ := validateAdminReq(ctx, w, r, policy.DecommissionAdminAction)
+	if objectAPI == nil {
+		return
+	}
+
+	// Legacy args style such as non-ellipses style is not supported with this API.
+	if globalEndpoints.Legacy() {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrNotImplemented), r.URL)
```

Comment:
Is ErrNotImplemented correct for this? Should we return BadRequest?

Reply:
We should return ErrInvalidArgument